src/main/java/io/cdap/plugin/http/common/BaseHttpConfig.java

```diff
@@ -314,113 +344,158 @@ public Boolean isServiceAccountFilePath() {
+        break;
+      case SERVICE_ACCOUNT:
+        String reasonSA = "Service Account is enabled";
+        assertIsSet(getServiceAccountType(), PROPERTY_NAME_SERVICE_ACCOUNT_TYPE, reasonSA);
+        boolean propertiesAreValid = validateServiceAccount(failureCollector);
+        if (propertiesAreValid) {
+          try {
+            AccessToken accessToken = OAuthUtil.getAccessToken(this);
+          } catch (Exception e) {
+            failureCollector.addFailure("Unable to authenticate given service account info. ",
+                    "Please make sure all infomation entered correctly")
+                .withStacktrace(e.getStackTrace());
+          }
+        }
+        break;
+      case BASIC_AUTH:
+        String reasonBasicAuth = "Basic Authentication is enabled";
+        if (!containsMacro(PROPERTY_USERNAME)) {
+          assertIsSetWithFailureCollector(getUsername(), PROPERTY_USERNAME, reasonBasicAuth,
+              failureCollector);
         }
-        if (bServiceAccountJson != null && bServiceAccountJson) {
-            if (!Optional.ofNullable(getServiceAccountJson()).isPresent()) {
-                collector.addFailure("Service Account JSON can not be empty.",
-                                "Please provide Service Account JSON.")
-                        .withConfigProperty(PROPERTY_NAME_SERVICE_ACCOUNT_JSON);
-            }
+        if (!containsMacro(PROPERTY_PASSWORD)) {
+          assertIsSetWithFailureCollector(getPassword(), PROPERTY_PASSWORD, reasonBasicAuth,
+              failureCollector);
         }
-        return collector.getValidationFailures().size() == 0;
+        break;
     }
+  }
 
-    public void validate(FailureCollector failureCollector) {
-        // Validate OAuth2 properties
-        if (!containsMacro(PROPERTY_OAUTH2_ENABLED) && this.getOauth2Enabled()) {
-            String reasonOauth2 = "OAuth2 is enabled";
-            assertIsSetWithFailureCollector(getTokenUrl(), PROPERTY_TOKEN_URL, reasonOauth2, failureCollector);
-            assertIsSetWithFailureCollector(getClientId(), PROPERTY_CLIENT_ID, reasonOauth2, failureCollector);
-            assertIsSetWithFailureCollector(getClientSecret(), PROPERTY_CLIENT_SECRET, reasonOauth2, failureCollector);
-            assertIsSetWithFailureCollector(getRefreshToken(), PROPERTY_REFRESH_TOKEN, reasonOauth2, failureCollector);
-        }
+  private void validateOAuth2Fields(FailureCollector failureCollector, String reasonOauth2) {
+    String reasonOauth2GrantType =
+        reasonOauth2 + " and grant type is " + getOauth2GrantType().getValue();
 
-        if (!containsMacro(PROPERTY_WAIT_TIME_BETWEEN_PAGES) && waitTimeBetweenPages != null
-          && waitTimeBetweenPages < 0) {
-            failureCollector.addFailure("Wait Time Between Pages cannot be a negative number.",
-              null).withConfigProperty(PROPERTY_WAIT_TIME_BETWEEN_PAGES);
+    switch (getOauth2GrantType()) {
+      case CLIENT_CREDENTIALS:
```

Comment:
common ones can be taken outside switch and just check for specific ones inside case. Only refresh token is something different, so we can add a if statement rather than switch and evaluate common ones before that